src/Latte/Macros/CoreMacros.php

```diff
@@ -869,12 +875,32 @@ public function macroVarPrint(MacroNode $node): string
 	/**
 	 * {templateType ClassName}
 	 */
-	public function macroTemplateType(MacroNode $node): void
+	public function macroTemplateType(MacroNode $node, PhpWriter $writer): string
 	{
 		if (!$this->getCompiler()->isInHead()) {
 			throw new CompileException($node->getNotation() . ' is allowed only in template header.');
 		}
 		$node->validate('class name');
+
+		$templateTypes = [];
+		try {
+			$reflectionClass = new ReflectionClass($node->args);
+			foreach ($reflectionClass->getProperties() as $property) {
```

Comment:
getProperties returns only real properties, it is not working for `@property` / `@property-read` annotation